### PR TITLE
Withdraw tests and add test ids for staking

### LINKED ISSUE #1217: E2E - Staking Functionality

### LINKED ISSUE #1218: Remove Stake NOW

### LINKED ISSUE #1241: Add data-testid's for withdrawals section

### LINKED ISSUE #1243: Add Ui tests for withdrawals page
add tests to check for withdrawals page

### LINKED ISSUE #1245: Remove stake at next epoch

### LINKED ISSUE #1246: Add eslint / linting to TFE automation

### PR DESCRIPTION
**Pull request by Ditmir-Vega**
closes #1241 
closes #1217 
closes #1218 
closes #1245
closes #1243 
closes #1246

### CODE REVIEW

File: automation/features/step-definitions/withdraw.steps.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Then(/^I enter a new wallet address "([^"]*)?"$/, (newWalletAddress) => {
+    WithdrawalsPage.ethWalletAddressField.setValue(newWalletAddress)
+});
+
+Then(/^the wallet address link is changed to "([^"]*)?"$/, (walletAddresslinkText) => {
+    expect(WithdrawalsPage.enableWalletAddressFieldLink).toHaveText(walletAddresslinkText)
+});
+
+Then(/^I can see the wallet address in the input field is "([^"]*)?"$/, (newWalletAddress) => {
+  console.log(WithdrawalsPage.ethWalletAddressField.getValue())
```
Comment (dexturr) [inline automation/features/step-definitions/withdraw.steps.js:32]:
```suggestion
```

File: automation/features/step-definitions/withdraw.steps.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const { Given, When, Then } = require("@wdio/cucumber-framework");
```
[the remark this answers is not in the document]
Comment (dexturr) [reply automation/features/step-definitions/withdraw.steps.js:1]: Can you configure your IDE to lint on save as well please adn install the eslint plugin if it is not there already

**Review by mattrussell36 — CHANGES_REQUESTED** [review]
Can you format everything consistently please. Also worth installing the recommended extensions so you don't have to do this manually.

File: automation/features/step-definitions/withdraw.steps.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ditmir-Vega) [reply automation/features/step-definitions/withdraw.steps.js:1]: we actually need to configure the eslint file for the automation folder as the old one was for cypress and hasn't been used . will do this for this pr

File: automation/features/step-definitions/withdraw.steps.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  expect(WithdrawalsPage.enableWalletAddressFieldLink).toHaveText('Enter address manually')
+  WithdrawalsPage.enableWalletAddressFieldLink.click()
+});
+
+Then(/^the wallet address field is enabled$/, () => {
+    expect(WithdrawalsPage.ethWalletAddressField).toBeEnabled()
+    expect(WithdrawalsPage.ethWalletAddressField).not.toHaveAttribute('disabled')
+});
+
+Then(/^I enter a new wallet address "([^"]*)?"$/, (newWalletAddress) => {
```
[the remark this answers is not in the document]
Comment (Ditmir-Vega) [reply automation/features/step-definitions/withdraw.steps.js:23]: in this case , new is a different wallet address than the one already filled into the field for withdrawals

File: automation/features/step-definitions/staking.steps.js [hunk 4]
```diff
@@ -1,132 +1,179 @@
...
+Then(/^the epoch countdown timer is counting down$/, () => {
+    const currentCountdownTimerText = browser.getByTestId('current-epoch-ends-in').getText()
+    browser.pause(2100) // let some time pass 
+    expect(browser.getByTestId('current-epoch-ends-in').getText()).not.toEqual(currentCountdownTimerText)
 })
 
-Then(/^I pause some seconds$/,()=>{
-browser.pause(600000)
+Then(/^I pause some seconds$/, () => {
+    browser.pause(600000)
```
Comment (jtsang586) [inline automation/features/step-definitions/staking.steps.js:56]: Unused step?

File: automation/features/step-definitions/staking.steps.js [hunk 5]
```diff
@@ -1,132 +1,179 @@
...
+    expect(stakingPage.removeStakeNowDisclaimerText.getText()).toEqual(disclaimerText)
+});
+
+Then(/^the submit button text is "([^"]*)?"$/, (btnText) => {
+    stakingPage.tokenAmountSubmitBtn.waitForEnabled()
+    expect(stakingPage.tokenAmountSubmitBtn).toHaveText(btnText)
+});
+
+Then(/^I can see the stake is removed immediately$/, () => {
+    $('.callout--success').waitForDisplayed({ timeout: 60000, timeoutMsg: "remove stake now success message container not displayed after 60 seconds" })
```
Comment (jtsang586) [inline automation/features/step-definitions/staking.steps.js:173]: There's quite a few steps here with timeouts of 20-60 seconds. Is it possible to reduce these? 
If anything takes more than 10 seconds to load that should probably be looked at. Understandable if some of these are waiting for epochs etc.

Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js:173]: this particulat step is waiting for the success of stake removal so might actually need to be longer , but we wont be able to do much anbout reducing these to 10 seconds until we get capsule installed

File: automation/features/step-definitions/staking.steps.js (hunk 4 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/staking.steps.js:56]: it is used for debugging but i have removed it here

**Review by jtsang586 — APPROVED** [review]

File: automation/features/step-definitions/withdraw.steps.js (hunk 1 above)
Comment (Ditmir-Vega) [reply automation/features/step-definitions/withdraw.steps.js]: addressed

**Comment by Ditmir-Vega** [comment]
@mattrussell36  I have addressed comments regarding linting and have added the automation files as part of the linting rules

**Comment by Ditmir-Vega** [comment]
merging pull request as all comments have been addresses

**Pull request merged**